theseus/core/robust_cost_function.py

```diff
@@ -165,3 +165,65 @@ def _supports_masking(self) -> bool:
     def _supports_masking(self, val: bool):
         self.cost_function._supports_masking = val
         self.__supports_masking__ = val
+
+class GNCRobustCostFunction(RobustCostFunction):
+    def __init__(
+        self,
+        cost_function: CostFunction,
+        loss_cls: Type[GNCRobustLoss],
+        log_loss_radius: Variable,
+        gnc_control_val: Variable,
+        name: Optional[str] = None,
+    ):
+        super().__init__(cost_function, loss_cls, log_loss_radius, name=name)
```

Comment:
We should check that `self.loss` is a subclass of `GNCRobustLoss` and throw `ValueError` with an explanation otherwise.